bindings/capacitorjs/ios/Plugin/Plugin.swift

```diff
@@ -0,0 +1,207 @@
+                  let secretManager = call.getObject("secretManager") else {
+                return call.reject("storagePath, clientOptions, coinType, and secretManager are required")
+            }
+            guard JSONSerialization.isValidJSONObject(clientOptions),
+                  JSONSerialization.isValidJSONObject(secretManager) else {
+                return call.reject("clientOptions or secretManager is an invalid JSON object")
+            }
+            let ClientOptions = try? JSONSerialization.data(withJSONObject: clientOptions)
+            // TODO: replacing for urls slashes temporaly, make better using Codable structs with URL type?
+            let stringfiedClientOptions = String(data: ClientOptions!, encoding: .utf8)!.replacingOccurrences(of: "\\", with: "")
+            
+            let SecretManager = try? JSONSerialization.data(withJSONObject: secretManager)
+            let stringfiedSecretManager = String(data: SecretManager!, encoding: .utf8)!
+            print(stringfiedSecretManager)
+            // prepare the internal app directory path
+            let fm = FileManager.default
+            guard let documents = fm.urls(
+                for: .applicationSupportDirectory,
+                   in: .userDomainMask
+            ).first else { 
+                return  call.reject("can not create the path")
+            }
+            let path = documents.appendingPathComponent(
+                storagePath,
+                isDirectory: true
+            ).path
+            
+            if !fm.fileExists(atPath: path) {
+                try fm.createDirectory(
+                    atPath: path,
+                    withIntermediateDirectories: true,
+                    attributes: nil
+                )
+            }
+            
+            // Exclude folder from auto-backup
+            var urlPath = URL(fileURLWithPath: path, isDirectory: true)
+            var values = URLResourceValues()
+            values.isExcludedFromBackup = true
+            try urlPath.setResourceValues(values)
+
+            // we need to modify the path on the JS object
+            let options = """
+            {
+                "storagePath":"\(path)",
+                "clientOptions":\(String(describing: stringfiedClientOptions)),
+                "coinType":\(coinType),
+                "secretManager":{
+                    "stronghold":{
+                        "snapshotPath":"\(path)/wallet.stronghold"
+                }}
+            }
+            """
+            print(options)
+            
+            // TODO: implement logger as a fn
+            let filename = "\(path)/iota_wallet.log"
+            let level_filter = "debug"
+            iota_init_logger(filename.cString(using: .utf8), level_filter.cString(using: .utf8))
+            
+            // Keep the C++ handler / pointer of the messageHandler call result
+            let handler: OpaquePointer? = iota_initialize(options, error_buffer, error_buffer_size)
```

Comment:
`options` looks like a Swift string, so do we need to pass `options.cString(using: .utf8)` like we did for line 119? Also, shouldn't we be checking (and possibly freeing) the `error_buffer`?

Reply:
Yes you're right, I missed that 😅 it's working the same, but I added it just in case.
Also I've put the error vars inside each function.

Reply:
Me too, seems it was in the swift docs in the past but was removed, cant find it in the new docs. 
https://stackoverflow.com/questions/41308409/converting-from-swift-string-to-const-char
```
When a function is declared as taking an UnsafePointer argument, it can accept any of the following:

        A String value, if Type is Int8 or UInt8. The string will automatically be converted to UTF8 in a buffer, and a pointer to that buffer is passed to the function.
```

Reply:
This looks like the closest current resource https://developer.apple.com/documentation/swift/using-imported-c-functions-in-swift. I wish they said it a bit more explicitly though